### PR TITLE
Remove Spread-TypeSpec

### PR DESCRIPTION
**Pull request by mcgallan**
Add test for Parameter/Spread, removed Spread-TypeSpec Mockapi and related tests.
For more detailed tasks, please refer to [this issue](https://github.com/Azure/autorest.csharp/issues/4846).

### CODE REVIEW

File: test/CadlRanchProjects.Tests/parameters-spread.cs [hunk 1]
```diff
@@ -33,8 +40,106 @@ public class ParametersSpreadTests : CadlRanchTestBase
...
+
+        [Test]
+        public Task Parameters_Spread_Alias_spreadAliasinAlias() => Test(async (host) =>
+        {
+            Response response = await new SpreadClient(host, null).GetAliasClient().SpreadParameterWithInnerAliasAsync("1", "bar", "foo", 1);
+            Assert.AreEqual(204, response.Status);
+        });
+
+        [Test]
+        public void SpreadModelMethod()
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/parameters-spread.cs:62]:
```suggestion
        public void SpreadAsRequestBody()
```

Comment (pshao25) [reply test/CadlRanchProjects.Tests/parameters-spread.cs:62]: Same as the below tests

Comment (mcgallan) [reply test/CadlRanchProjects.Tests/parameters-spread.cs]: fixed

**Review by pshao25 — APPROVED** [review]

**Comment by pshao25** [comment]
Attach the issue.

**Pull request merged**